lib/metadataTypes/Folder.js

```diff
@@ -299,30 +298,29 @@ class Folder extends MetadataType {
      * @returns {Promise} Promise
      */
     static async update(metadata) {
-        return await new Promise((resolve) => {
-            const tempCustomerKey = metadata.CustomerKey;
-            const path = metadata.Path;
-            this.removeNotUpdateableFields(metadata);
-            this.client.SoapClient.update('DataFolder', metadata, null, (ex, response) => {
-                if (ex && ex.results) {
-                    Util.logger.error(
-                        `${this.definition.type}-${metadata.ContentType}.update:: error updating: '${path}'. ${ex.results[0].StatusMessage}`
-                    );
-                    resolve();
-                } else if (ex) {
-                    Util.logger.error(
-                        `${this.definition.type}-${metadata.ContentType}.update:: error updating: '${path}'. ${ex.message}`
-                    );
-                    resolve();
-                } else {
-                    response.body.Results[0].Object = metadata;
-                    response.body.Results[0].Object.CustomerKey = tempCustomerKey;
-                    delete response.body.Results[0].Object.$;
-                    Util.logger.info(`- updated folder: ${path}`);
-                    resolve(response);
-                }
-            });
-        });
+        const tempCustomerKey = metadata.CustomerKey;
+        const path = metadata.Path;
+        let response;
```

Comment:
ditto

Reply:
- [ ] turn `let` into `const` inside of try-catch